universal-application-tool-0.0.1/app/assets/javascripts/number.ts

```diff
@@ -8,8 +8,9 @@ class NumberController {
       <NodeListOf<HTMLInputElement>>document.querySelectorAll('input[type=number]'));
     numberQuestions.forEach(numberQuestion => {
       numberQuestion.addEventListener('input', () => {
-        // Only allow integer input.
-        const n = parseInt(numberQuestion.value.replace(/[^.\d]/g, ''));
+        // Block non-numeric characters like like 'e', 'E', '+', '-'
+        // which will cause problems when parsing to a number
+        const n = parseFloat(numberQuestion.value.replace(/[^.\d]/g, ''));

```

Comment:
I think we probably want a validation.ts method to show an error message rather than manipulating user key strokes. This is similar to currency and Address Zip code validation where we validate upon Submit.

Reply:
app/assets/javascripts/validation.ts

Reply:
Do we still need this?  All the validation should be a submit not typing time.

Reply:
I think we still don't need this BTW.